### PR TITLE
apply search field

### LINKED ISSUE #5: Input Component
## Expected Behaviour

Please write an input component which is usable on different places. Because the Input Component doesnt look always the same. You should be able pass some props to the input field.

## Checklist

- [x] Create a component folder if not existing
- [x]  Create a Input.tsx Component
- [x] follow the style from [Figma](https://www.figma.com/design/TOCJzJFMXE7PL1dQyGWqXZ/midterm?node-id=193-51&t=auNq1TlniIu9XziH-0)
- [x] Make sure that you forward all "attributes" to your input field that an input field can have.
- [x] make sure that you can pass a Icon to your component (Icon is optional)
- [x]  make sure that you can change the border radius to full rounded

### Hints

    - If you style the label like an input field it's easier to place the icon in the right position.
    - If you look for an Icon maybe use https://react-icons.github.io/react-icons/

### PR DESCRIPTION
**Pull request by Hydrateurself**
## Checklist

Confirm you have completed the following actions prior to submitting this PR.

- [ ] There is an existing issue report for this PR.
- [ ] I have created a feature branch.
- [ ] My changes have been committed.
- [ ] I have pushed my changes to the branch.

## Title

_Give your PR a short title summarising the patch, bug fix or feature._

## Description

_Ensure the PR description clearly describes the problem and solution and provide as much relevant information as possible._

## Issue Resolution

_Tell us which issue this PR fixes._

Closes #5

### CODE REVIEW

File: src/components/input.tsx [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+                   
+            <label className="flex items-center w-[335px] h-[48px] rounded-full bg-dark-light border-2 border-dark-light focus-within:border-white-dimmed-heavy ">
+
+                <div className=" w-auto h-auto ml-5 absolute flex">
+                    <IoIosSearch className="text-white-dimmed h-5" />
+                </div>
+                
+                             <input type="text"
+                                 placeholder= {placeholder}
+                                 className="bg-transparent w-[335px] h-[48px] rounded-full placeholder: font-medium placeholder: font-inter placeholder: text-sm placeholder: leading-4 placeholder: pl-12 outline-none text-white-dimmed" />
```
Comment (dnmct) [inline src/components/input.tsx]: some of the classes are not applied because you have spaces in there

File: src/components/input.tsx
Comment (dnmct) [inline src/components/input.tsx:1]: pls always format your files before pushing

File: src/components/input.tsx [hunk 2]
```diff
@@ -0,0 +1,34 @@
+import { IconType } from "react-icons"
+import { IoIosSearch } from "react-icons/io";
+
+type Props = {
+    placeholder: string,
+    icon: IconType,
```
Comment (dnmct) [inline src/components/input.tsx]: the icon prop is not used

File: src/components/input.tsx [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    iconColor: string
+}
+
+export const InputFields = ({ placeholder,icon}: Props) => {
```
Comment (dnmct) [inline src/components/input.tsx]: maybe just call it Input, every instance of a component is a single thing

File: src/components/input.tsx [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    
+    return (
+        
+      
+                   
+            <label className="flex items-center w-[335px] h-[48px] rounded-full bg-dark-light border-2 border-dark-light focus-within:border-white-dimmed-heavy ">
+
+                <div className=" w-auto h-auto ml-5 absolute flex">
+                    <IoIosSearch className="text-white-dimmed h-5" />
```
Comment (dnmct) [inline src/components/input.tsx]: use size-5 instead of h-5, to make sure aspect ratio is always square

File: src/components/input.tsx [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+    iconColor: string
+}
+
+export const InputFields = ({ placeholder,icon}: Props) => {
+    
+    return (
+        
+      
+                   
+            <label className="flex items-center w-[335px] h-[48px] rounded-full bg-dark-light border-2 border-dark-light focus-within:border-white-dimmed-heavy ">
```
Comment (dnmct) [inline src/components/input.tsx]: don't use arbitrary values for height and width work with padding instead

**Review by dnmct — CHANGES_REQUESTED** [review]

**Comment by Hydrateurself** [comment]
fixed the issues ready to merge?

**Review by dnmct — APPROVED** [review]

**Pull request merged**